pkg/descheduler/strategies/pod_lifetime.go

```diff
@@ -31,33 +31,48 @@ import (
 	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
 )
 
-func validatePodLifeTimeParams(params *api.StrategyParameters) error {
+var (
+	podLifeTimeAllowedStates = sets.NewString(
+		string(v1.PodRunning),
+		string(v1.PodPending),
+
+		// Container state reason list: https://github.com/kubernetes/kubernetes/blob/release-1.24/pkg/kubelet/kubelet_pods.go#L76-L79
+		"PodInitializing",
+		"ContainerCreating",
+	)
+)
+
+func validatePodLifeTimeParams(params *api.StrategyParameters) (sets.String, error) {
 	if params == nil || params.PodLifeTime == nil || params.PodLifeTime.MaxPodLifeTimeSeconds == nil {
-		return fmt.Errorf("MaxPodLifeTimeSeconds not set")
+		return nil, fmt.Errorf("MaxPodLifeTimeSeconds not set")
 	}
 
+	var states []string
 	if params.PodLifeTime.PodStatusPhases != nil {
-		for _, phase := range params.PodLifeTime.PodStatusPhases {
-			if phase != string(v1.PodPending) && phase != string(v1.PodRunning) {
-				return fmt.Errorf("only Pending and Running phases are supported in PodLifeTime")
-			}
-		}
+		states = append(states, params.PodLifeTime.PodStatusPhases...)
+	}
+	if params.PodLifeTime.States != nil {
+		states = append(states, params.PodLifeTime.States...)
+	}
+	if !podLifeTimeAllowedStates.HasAll(states...) {
+		return nil, fmt.Errorf("states must be one of %v", podLifeTimeAllowedStates.List())
```

Comment:
nit: this would be nice if it said which state was invalid, but that can be a todo for later

Reply:
that's a good idea, PR coming up soon